src/internet_identity/tests/integration/http.rs

```diff
@@ -268,6 +270,120 @@ fn should_set_cache_control_for_spa() -> Result<(), CallError> {
     Ok(())
 }
 
+/// Verifies that the cache-control header is set for the icons.
+#[test]
+fn should_set_cache_control_for_icons() -> Result<(), CallError> {
+    const CERTIFICATION_VERSION: u16 = 2;
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+
+    // Get index page
+    let index_request = HttpRequest {
+        method: "GET".to_string(),
+        url: "/".to_string(),
+        headers: vec![],
+        body: ByteBuf::new(),
+        certificate_version: Some(CERTIFICATION_VERSION),
+    };
+    let index_response = http_request(&env, canister_id, &index_request)?;
+
+    // Convert body to string and find the font URL
+    let index_html =
+        String::from_utf8(index_response.body.into_vec()).expect("Failed to parse HTML");
+
+    // Find the spa URL in the HTML
+    let spa_url = {
+        let spa_suffix = "cacheable.js";
+        let spa_end = index_html
+            .find(spa_suffix)
+            .expect("Could not find cacheable.js in HTML");
+        let prefix_start = index_html[..spa_end]
+            .rfind('/')
+            .expect("Could not find starting / for spa URL");
+
+        index_html[prefix_start..spa_end + spa_suffix.len()].to_string()
+    };
+
+    // Get spa file
+    let spa_request = HttpRequest {
+        method: "GET".to_string(),
+        url: spa_url,
+        headers: vec![],
+        body: ByteBuf::new(),
+        certificate_version: Some(CERTIFICATION_VERSION),
+    };
+
+    let spa_response = http_request(&env, canister_id, &spa_request)?;
+
+    assert_eq!(spa_response.status_code, 200);
+    assert!(spa_response.headers.contains(&(
+        "Cache-Control".to_string(),
+        "public, max-age=31536000".to_string()
+    )));
+
+    let spa_result = verify_response_certification(
+        &env,
+        canister_id,
+        spa_request,
+        spa_response.clone(),
+        CERTIFICATION_VERSION,
+    );
+    assert_eq!(spa_result.verification_version, CERTIFICATION_VERSION);
+
+    let spa_bytes = spa_response.body.into_vec();
+
+    // Decompress the spa bytes
+    let mut decoder = GzDecoder::new(&spa_bytes[..]);
+    let mut spa_body = String::new();
+    decoder.read_to_string(&mut spa_body).unwrap();
+
+    // Find the icon URL in the HTML
+    let icon_url = spa_body
+        .lines()
+        .find(|line| line.contains("icpswap_logo"))
```

Comment:
Can you make `"icpswap_logo"` a variable and use it in both places?

It would be good to have clearly which one we are selecting and easily change it.

Reply:
done